autogalaxy/analysis/chaining_util.py

```diff
@@ -230,35 +234,243 @@ def clumps_from(
     return clumps
 
 
-def mass_light_dark_from(
-    lmp_model: af.Model(LightMassProfile),
-    result_light_component: af.Model,
-) -> Optional[af.Model]:
+def mass_light_dark_lmp_from(
+    light_result: Result, name : str, light_is_model : bool = False
+):
     """
-    Returns an updated version of a `LightMassProfile` model (e.g. a bulge or disk) whose priors are initialized from
-    previous results of a `Light` pipeline.
+    Returns a light and mass profile from a standard light profile (e.g. a Sersic) in the LIGHT PIPELINE result, for
+    the LIGHT DARK MASS PIPELINE.
 
-    This function generically links any `LightProfile` to any `LightMassProfile`, pairing parameters which share the
-    same path.
+    For example, if the light pipeline fits a Sersic profile, this function will return a Sersic profile for the LIGHT
+    DARK MASS PIPELINE.
+
+    If a linear light profile is passed in, the function will return a light and mass profile where its `intensity` is
+    fixed to the solved for value of the maximum log likelihood linear fit.
+
+    The light and mass profiles can also be switched to variants which have a radial gradient in their mass-to-light
+    conversion, by setting the `include_mass_to_light_gradient` parameter to `True`.
 
     Parameters
     ----------
-    lmp_model
-        The light and mass profile whoses priors are passed from the LIGHT PIPELINE.
-    result_light_component
-        The `LightProfile` result of the LIGHT PIPELINE used to pass the priors.
+    light_result
+        The result of the light pipeline, which determines the light and mass profiles used in the LIGHT DARK
+        MASS PIPELINE.
+    name
+        The name of the light profile in the light pipeline's galaxy model that the model is being created for
+    light_is_model
+        If `True`, the light profile is passed as a model component, else it is a fixed instance.
 
     Returns
     -------
-    af.Model(mp.LightMassProfile)
-        The light and mass profile whose priors are initialized from a previous result.
+    The light and mass profile for a standard light profile whose priors are initialized from a previous result.
     """
 
-    if lmp_model is None:
-        return lmp_model
+    lp_instance = getattr(light_result.instance.galaxies.lens, name)
+    lp_model = getattr(light_result.model.galaxies.lens, name)

```

Comment:
I wonder if it's worth handling an attribute error and raising a new attribute error with an informative message about the form the model must take?